main/core/src/eval/Evaluator.scala

```diff
@@ -257,22 +270,24 @@ case class Evaluator(home: os.Path,
     }
   }
 
-  def evaluateGroup(group: Agg[Task[_]],
-                    results: collection.Map[Task[_], Result[(Any, Int)]],
-                    inputsHash: Int,
-                    paths: Option[Evaluator.Paths],
-                    maybeTargetLabel: Option[String],
-                    counterMsg: String,
-                    reporter: Int => Option[BuildProblemReporter],
-                    testReporter: TestReporter,
-                    logger: Logger): (mutable.LinkedHashMap[Task[_], Result[(Any, Int)]], mutable.Buffer[Task[_]]) = {
-
+  protected def evaluateGroup(group: Agg[Task[_]],
+    results: collection.Map[Task[_], Result[(Any, Int)]],
+    inputsHash: Int,
+    paths: Option[Evaluator.Paths],
+    maybeTargetLabel: Option[String],
+    counterMsg: String,
+    reporter: Int => Option[BuildProblemReporter],
+    testReporter: TestReporter,
+    logger: Logger
+   ): (mutable.LinkedHashMap[Task[_], Result[(Any, Int)]], mutable.Buffer[Task[_]]) = PrintLogger.withContext(maybeTargetLabel.map(_ + ": ")) {
 
     val newEvaluated = mutable.Buffer.empty[Task[_]]
     val newResults = mutable.LinkedHashMap.empty[Task[_], Result[(Any, Int)]]
 
     val nonEvaluatedTargets = group.indexed.filterNot(results.contains)
 
+    var logEnd: Option[() => Unit] = None
```

Comment:
Can we make `logEnd` a `val`, and just initialize it immediately to `None` or `Some` depending on `effectiveThreadCount`?

Reply:
Changed. I will probably ditch the end logging. We have the more granular debug log if interested in end events.